File: cases/trees/api-be/services/be-reliability.console-in-loop.go
// be-reliability/console-in-loop (Go lane) — bad: a fmt write the parser places INSIDE a for
// statement's projected span, so it runs once per iteration. good: one aggregated write after the
// loop — that line still co-fires console-in-be (this file sits under services/), which is a
// different claim (WHERE the write is, not how many times it runs); same pairing as the TS fixture.
package services

import "fmt"

func BadLoop(orderIds []string) {
	for _, id := range orderIds {
		fmt.Println("processing order " + id)
	}
}

func GoodAfter(orderIds []string) {
	for _, id := range orderIds {
		accumulate(id)
	}
	fmt.Println(len(orderIds))
}

func accumulate(id string) {}

File: cases/trees/api-be/services/be-reliability.console-in-loop.cs
// be-reliability/console-in-loop (C# lane) — bad: a Console write the parser places INSIDE a foreach
// statement's projected span, so it runs once per iteration. good: one aggregated write after the
// loop — that line still co-fires console-in-be (this file sits under services/), a different claim;
// same pairing as the TS and Go fixtures.
class BeReliabilityConsoleInLoop {
  void Bad(string[] orderIds) {
    foreach (var id in orderIds) {
      Console.WriteLine("processing order " + id);
    }
  }

  void Good(string[] orderIds) {
    foreach (var id in orderIds) {
      Accumulate(id);
    }
    Console.WriteLine(orderIds.Length);
  }

  void Accumulate(string id) {}
}
